### PR TITLE
Test dapp and its tests

### PR DESCRIPTION
**Pull request by gaetbout**
Adding the test_dapp contract used for testing other contracts

### CODE REVIEW

**Review by Leonard-Pat — APPROVED** [review]

**Comment by juniset** [comment]
Do we really need unit tests on the test dapp? The test dapp is used to test calls from the account, but it serves no purpose in itself. Not sure I see value in these tests.

**Comment by gaetbout** [comment]
Fine for me to remove them, so as you want.
I mostly made them check some felt overflow stuff with Cairo 1.0
(plot twist exact same as Cairo 0.x)

**Comment by juniset** [comment]
I'm fine either way. On one side they're already there so might as well keep them, on the other side it means that we will need to maintain them for no benefit. 

Your call.

File: contracts/test_dapp.cairo [hunk 1]
```diff
@@ -0,0 +1,59 @@
+#[contract]
+mod TestDapp {
+    // use array::ArrayTrait;
+    // use contracts::asserts;
```
Comment (delaaxe) [inline contracts/test_dapp.cairo:4]: no commented code

**Review by delaaxe — APPROVED** [review]

**Review by juniset — APPROVED** [review]

**Pull request merged**